rust/agents/relayer/src/main.rs

```diff
@@ -38,7 +38,9 @@ async fn _main() -> Result<()> {
 
     let _ = agent.metrics().run_http_server();
 
-    agent.run().await??;
+    let all_fut_tasks = agent.run().await?;
+    all_fut_tasks.await??;
```

Comment:
I like this approach to making it a bit more clean.

Reply:
Seemed less bad than the alternative, though I still find myself stuttering when reading a double-question-mark code snippet